[Kan Jitpakdi] iP

ryukanjit opened the pull request:
Welcome to my pull request fellow CS2103T classmates 🦁 

# Welcome to Alice!
## _The Last Virtual Assistant You will ever need, Ever_


Alice is a virtual assistant started off as Duke task manager before evolving in to her own form.

- Choose a save file by typing the save file name in
- See your current task
- ✨Start editing✨

## Features

- [X] Create local save files location for Alice to remember the task
- [X] Fetch certain deadlines and other tasks using the yyyy-MM-dd format
- [X] Fetch certain deadlines and other tasks using "keywords"
- [X] GUI 
- [x] Create, Save, Delete save files
- [ ] Integration with Website (Coming Soon)

> Let's make a slight change to create a big difference

alice.Alice is aimed to be implmeneted into a website in the future with 
continuously improving UI, UX, and backend.
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(alice.Alice.class, args);
    }
}
```
![alice_0.03](https://github.com/kanjitp/ip/blob/master/src/main/resources/screenshots/alice_0.03_demo.gif?raw=true)

Commit:
Add Gradle support

ryukanjit committed:
Level-1

ryukanjit committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d'

ryukanjit committed:
Level-2: allow more methods in Dialog.java and change Duke.java to match level 2 requirement

ryukanjit committed:
Level-3: add Task.java, TaskDialog.java (extends Dialog.java), and change Duke.java (commands dialog), Dialog.java (protected sentences, archive)

ryukanjit committed:
Level-4: add Todo.java, Event.java, Deadline.java, minor typo correction in command window

ryukanjit committed:
A-TextUiTesting: Adjust Dialog.java, Duke.java, TaskDialog.java  (no extra space to avoid bugs)

ryukanjit committed:
Level-5: add Exception handling - add DialogException.java, DukeException.java, EmptyDescriptionException.java, EmptyIndexException.java, EmptyTaggerException.java, InvalidArgumentException.java, InvalidIndexException.java, InvalidIndexException.java

ryukanjit committed:
Level-5: add testing

ryukanjit committed:
Level-6: add method deleteTaskByIndex(int index) to TaskDialog, adjust Duke.java, add testing

ryukanjit committed:
A-Enums: add enum Command, and update Duke.java

ryukanjit committed:
Add save functionality, and new test and files: input2.txt, EXPECTED2.TXT

ryukanjit committed:
Add TimeTask.java TimeFormatException.java, Refactor Deadline.java Event.java Duke.java

ryukanjit committed:
Add date filter functionality - update Duke.java, add method by(String deadlineString) to TaskDialog.java

ryukanjit committed:
Add description to Duke.java

This commit is just a dummy mainly for testing out to see the shape of git tree for educational purpose

ryukanjit committed:
Merge branch 'branch-Level-8'

ryukanjit committed:
Merge branch 'branch-Level-7'

add new testing
fix testing bug

ryukanjit committed:
Refactor Duke into different classes: Ui.java, Storage.java, Parser.java, TaskList.java, Command.java - Duke.java is now Alice.java

ryukanjit committed:
Divide classes into packages

ryukanjit committed:
Add test for Dialog.java, remove replicated method

#Replicated methods
have(String id) and containsId(String id) - keep containsId(String id)
getDialog(String id) and get(String key) - rename and keep get(String id)

ryukanjit committed:
Add jar build modification

#update
Storage can now build the save location under ./alice/data/
Ui refactored to use Storage methods when initially asking the user to choose save file

ryukanjit committed:
Add JavaDoc documentation to every classes from branching point

ryukanjit committed:
Add find functionality to Alice

ryukanjit committed:
Merge branch 'branch-Level-9'

ryukanjit committed:
Tweak the code to comply with CS2103T coding standard

ryukanjit committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/dialog/TaskDialog.java

ryukanjit committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/command/Command.java
#	src/main/java/parser/Parser.java

ryukanjit committed:
Update project README.md

timiditi commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,195 @@
+            }
+            break;
+
+        case TODO: {
```
Are we supposed to put case statements in curly brackets in switch statements?

kjpvi replied:
Fixed thanks a lot :))

timiditi commented on src/main/java/dialog/TaskDialog.java:
```diff
@@ -0,0 +1,193 @@
+     * @return TaskDialog with new TaskList stored in it
+     * @throws InvalidTimeFormatException time of illegal format (not yyyy-MM-dd)
+     */
+    public TaskDialog by(String deadline) throws InvalidTimeFormatException {
```
Do you want to rename this method to be a verb? e.g. getFromDeadline?

kjpvi replied:
Yeah, most likely a good idea I was at that time thinking of stream chaining so I named it in such manner

timiditi commented on src/main/java/dialog/TaskDialog.java:
```diff
@@ -0,0 +1,193 @@
+     * @param keyword the keyword to search description
+     * @return TaskDialog with TaskList with tasks matching the keyword
+     */
+    public TaskDialog with(String keyword) {
```
I would suggest renaming this method to have a verb for a name too, perhaps findFromKeyword.

kjpvi replied:
Same as previous comment

timiditi commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,81 @@
+     *
+     * @return description of the task
+     */
+    public String description() {
```
I think you should rename this to getDescription to fit with the coding standard.

kjpvi replied:
fixed

timiditi commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,126 @@
+     *
+     * @return the length of the tasks
+     */
+    public int length() {
```
Want to rename this to getLength()?

kjpvi replied:
will change in later commits

timiditi reviewed (commented):
LGTM, just a few nitpicks on coding standards. Nice work!

ryukanjit committed:
Add GUI using JavaFx

- migrate Alice to alice folder
- Alice now doesn't have a main class but extends Application from JavaFX

ryukanjit committed:
Add JavaFx Gradle Support

ryukanjit committed:
Merge branch 'add-gradle-support'

ryukanjit committed:
Merge branch 'branch-Level-10'

ryukanjit committed:
Add Java Doc for current build and short demo in main/resources

ryukanjit committed:
Fix Alice to use fxml

ryukanjit committed:
Adjust config to allow build by shadowJar

ryukanjit committed:
Add delete functionality to Alice

User could not delete their save files before so now they would be able to interact with their data with more functionality

Change button view to list view instead to integrate with delete functionality

Delete will remove the file name from the list view but the file won't actually be deleted till the user move to new page.
Therefore, the method for handling deletion of file needs to check the file names from the list view not the files in the
directory

Other things done in this commit:
* Add JavaDoc to new methods
* Add JavaDoc to new Class AlertBox
* Add video demo of the current build

ryukanjit committed:
Update README

Previous README does not reflect the current build of the chatbot

Add what more features done and video demo

The video demo should give the repo a better impression to visitors

ryukanjit committed:
Use assert in non-custom exceptions

some exceptions can be easier deal with AssertionException

### Other changes things included
- [x] separate exceptions into sub-folder exceptions inside each folder
- [x] fix bug where the same task is added but Alice doesn't print error to the screen

ryukanjit committed:
Improve code quality and refactor to improve readability and abstraction.

Long methods are now divided into smaller methods with better abstraction.

### Highlight
- Improve logic flow of AlertBox
- Separate different commands into multiple private methods to make handling more logical and easier to navigate to different parts

ryukanjit committed:
Fix javaDoc and indentations.

For better file formatting.

ryukanjit committed:
Merge pull request #2 from kanjitp/branch-A-Assertions

Use assert in non-custom exceptions

ryukanjit committed:
Merge branch 'master' of github.com:kanjitp/ip

ryukanjit committed:
Merge pull request #3 from kanjitp/branch-A-CodeQuality

Branch a code quality

ryukanjit committed:
Merge branch 'master' of github.com:kanjitp/ip

ryukanjit committed:
Refactor the main logic of the app.

The previous versions contain many too convoluted flow of the program which highly decrease the efficiency of the program

ryukanjit committed:
Repackage task to be inside of model and add Vocab classes

task package originally exist on the same level as other package, now it is inside model package for better orgnaization

ryukanjit committed:
Add learn and unlearn commands for the user to teach Alice or make Alice forget about certain custom commands.

Alice would now be able to respond to user with specific phrase once the user have taught Alice what to respond.
This could prove to be useful for user to teach Alice a specific phrase and to respond accordingly.

The update also comes with unlearn functionality to make Alice more intuitively usable.

Note:
- User cannot teach Alice the command that Alice already know as Alice will respond back that the vocabulary user is trying to teach is untouchable
- User cannot teach the vocabulary that Alice already knows
- User cannot unteach/make Alice unlearn a vocabulary that she does not recognize or is a default command

ryukanjit committed:
Merge pull request #4 from kanjitp/branch-D-Trivia

Branch D Trivia PR

ryukanjit committed:
Merge branch 'master' of github.com:kanjitp/ip

ryukanjit committed:
Fix unlearn bug and readjust Start Page Ui

Previously unlearn would not actually delete the vocabulary from Alice dictionary.
Start Page now have a more symmetric layout

ryukanjit committed:
Improve the GUI, add Center Box, and fix typos

The GUI is now improved to be much more engaging improving the UI/UX when using the application.

Bugs found:
- Alice sometimes cannot throw DialogException

ryukanjit committed:
Add Acknowledgement to README and all photos are now licensed and royalty free

The previous photos breach copyright, now the app runs without concern for copyright.

ryukanjit committed:
Add and fix JavaDoc for every classes, and use check style, fix Alice bug when using date method.

The previous JavaDoc was out of date and contain many irrelevant and incorrect information about different classes and method.

Some JavaDoc also does not contain enough information for it to be useful or informative.

The JavaDoc now conforms to the coding standard specified by the module.

ryukanjit committed:
Minor refactoring and fix bug with addTask method and creating a new file will always result in task_list save file.

Add task used TaskDialog to generate the dialog instead which proves to be wrong usage.

ryukanjit committed:
Occupy REAME in docs with User Guide and add multiple screenshots to repo.

The previous versions does not contain screenshots to be used for README writing due to lack of URL.

Start building website.

ryukanjit committed:
Fix website README, new Ui.png, fix Dialog Test, and add TaskDialogTest.

Multiple house cleaning stuffs to make the project cleaner overall.

ryukanjit committed:
Fix bug: adding tasks of different type result in failure despite it being different task.

The previous version causes adding multiple tasks of the same name but different date or different type to result in a failure.

The app should function normally now and task of same name but different date or different type is now able to be contained in the same list.

ryukanjit committed:
Add demo gifs to repo to be used for README and User Guide

ryukanjit committed:
Update README and User Guide.

Now the reader can see the updated version of README and User Guide.

ryukanjit committed:
Minor README and UserGuide formatting.

ryukanjit committed:
Fix README 2.

ryukanjit committed:
Set theme jekyll-theme-cayman

ryukanjit committed:
Set theme jekyll-theme-tactile

ryukanjit committed:
Update UG.

ryukanjit committed:
Merge commit '96ac2c541dc94ad6268484face9d9455231abc2b'

ryukanjit committed:
Bug Fix for macOS and gif refactoring.

macOS previously would not show the correct text if the device doesn't have the specified text.

Refactor the code such that the screenshots are now in docs and not resources.

ryukanjit committed:
Update README.md
